sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -0,0 +1,629 @@
+import six
+
+from uamqp import ReceiveClient, Source, types
+from uamqp.constants import SenderSettleMode
+
+from ._base_handler import BaseHandler
+from ._common.utils import create_authentication
+from ._common.message import PeekMessage, ReceivedMessage
+from ._common.constants import (
+    REQUEST_RESPONSE_GET_SESSION_STATE_OPERATION,
+    REQUEST_RESPONSE_SET_SESSION_STATE_OPERATION,
+    REQUEST_RESPONSE_RENEW_SESSION_LOCK_OPERATION,
+    REQUEST_RESPONSE_RECEIVE_BY_SEQUENCE_NUMBER,
+    REQUEST_RESPONSE_UPDATE_DISPOSTION_OPERATION,
+    REQUEST_RESPONSE_RENEWLOCK_OPERATION,
+    REQUEST_RESPONSE_PEEK_OPERATION,
+    ReceiveSettleMode,
+    NEXT_AVAILABLE,
+    SESSION_LOCKED_UNTIL,
+    DATETIMEOFFSET_EPOCH,
+    SESSION_FILTER,
+    REQUEST_RESPONSE_RENEWLOCK_OPERATION
+)
+from .exceptions import (
+    _ServiceBusErrorPolicy,
+    SessionLockExpired
+)
+from ._common import mgmt_handlers
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ServiceBusSession(object):
+    """
+    The ServiceBusSession is used for manage session states and lock renewal.
+
+    **Please use the instance variable `session` on the ServiceBusReceiver to get the corresponding ServiceBusSession
+    object linked with the receiver instead of instantiating a ServiceBusSession object directly.**
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
+            :start-after: [START get_session_sync]
+            :end-before: [END get_session_sync]
+            :language: python
+            :dedent: 4
+            :caption: Get session from a receiver
+    """
+    def __init__(self, session_id, receiver, encoding="UTF-8"):
+        self._session_id = session_id
+        self._receiver = receiver
+        self._encoding = encoding
+        self._session_start = None
+        self._locked_until = None
+        self.auto_renew_error = None
+
+    def _can_run(self):
+        if self.expired:
+            raise SessionLockExpired(inner_exception=self.auto_renew_error)
+
+    def get_session_state(self):
+        # type: () -> str
+        """Get the session state.
+
+        Returns None if no state has been set.
+
+        :rtype: str
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
+                :start-after: [START get_session_state_sync]
+                :end-before: [END get_session_state_sync]
+                :language: python
+                :dedent: 4
+                :caption: Get the session state
+        """
+        self._can_run()
+        response = self._receiver._mgmt_request_response_with_retry(  # pylint: disable=protected-access
+            REQUEST_RESPONSE_GET_SESSION_STATE_OPERATION,
+            {'session-id': self.session_id},
+            mgmt_handlers.default
+        )
+        session_state = response.get(b'session-state')
```

Comment:
It seems a bit silly suggesting this for such a simple string, but should we const-ify keywords in the core mgmt request/response flow, or do you think that's overkill?  Am very open to thoughts on this.

Reply:
updated!